docs/configure/compilers.md

```diff
@@ -0,0 +1,163 @@
+---
+title: 'Compiler support'
+---
+
+Javascript compilers are essential in optimizing and transforming code, enhancing performance, and ensuring compatibility across different environments. Storybook provides support for the leading compilers, ensuring lightning-fast build time and execution with [SWC](https://swc.rs/) or leveraging [Babel](https://babeljs.io/) with its extensive ecosystem of plugins and presets to allow you to use the latest features of the ecosystem with minimal configuration required.
```

Comment:
We should make it clear in the very beginning that this page is just about configuring compilers for Webpack5-based projects. This whole page doesn't apply for vite Projects for example.